GoRogue/MapGeneration/Region.cs

```diff
@@ -1,355 +1,478 @@
+         public static Region ParallelogramFromTopCorner(Point origin, int width, int height)
+         {
+             Point nw = origin;
+             Point ne = origin + new Point(width, 0);
+             Point se = origin + new Point(width * 2, height);
+             Point sw = origin + new Point(width, height);
+
+             return new Region(nw, ne, se, sw);;
+         }
+
+         /// <summary>
+         /// Creates a new Region in the shape of a parallelogram, with diagonals going up and right.
+         /// </summary>
+         /// <param name="origin">Origin of the parallelogram.</param>
+         /// <param name="width">The horizontal length of the top and bottom sides.</param>
+         /// <param name="height">Height of the parallelogram.</param>
+         /// <returns>A new region in the shape of a parallelogram</returns>
+         public static Region ParallelogramFromBottomCorner(Point origin, int width, int height)
+         {
+             var negative = Direction.YIncreasesUpward ? 1 : -1;
+
+             Point nw = origin + (height, height * negative);
+             Point ne = origin + (height + width, height * negative);
+             Point se = origin + (width, 0);
+             Point sw = origin;
+
+             return new Region(nw, ne, se, sw);
+         }
+        #endregion
+
+        /// <summary>
+        /// Returns a string detailing the region's corner locations.
+        /// </summary>
+        public override string ToString()
+            => $"Region: NW{NorthWestCorner.ToString()}=> NE{NorthEastCorner.ToString()}=> SE{SouthEastCorner.ToString()}=> SW{SouthWestCorner.ToString()}";
+
+        /// <summary>
+        /// Is this Point one of the corners of the Region?
+        /// </summary>
+        /// <param name="here">the point to evaluate</param>
+        public bool IsCorner(Point here) =>
+            here == NorthEastCorner || here == NorthWestCorner || here == SouthEastCorner || here == SouthWestCorner;
+
+
+        #region Transformation
+
+        /// <summary>
+        /// Moves the Region in the indicated direction.
+        /// </summary>
+        /// <param name="x">The X-value by which to shift the region</param>
+        /// <param name="y">The Y-value by which to shift the region</param>
+        /// <returns></returns>
+        public virtual Region Translate(int x, int y)
+            => Translate((x, y));
+
         /// <summary>
-        /// Removes (from this region) the outer points shared in common
-        /// with the imposing region
+        /// Moves the region in the indicated direction.
         /// </summary>
-        /// <param name="imposing">the region to check for common outer points</param>
-        public void RemoveOverlappingPoints(Region imposing)
+        /// <param name="by">The amount (X and Y) to translate this region by.</param>
+        /// <returns>A new, translated region</returns>
+        public virtual Region Translate(Point by)
```

Comment:
Ditto above on if this needs to be virtual.

Also the corresponding functions in `Rectangle`/`Point` use name `delta` or `deltaChange` for parameter (don't recall).